src/plugins/vis_type_table_new/public/table_vis_renderer.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { render, unmountComponentAtNode } from 'react-dom';
+
+import { CoreStart } from 'opensearch-dashboards/public';
+import { VisualizationContainer } from '../../visualizations/public';
+import { ExpressionRenderDefinition } from '../../expressions/common/expression_renderers';
+import { TableVisRenderValue } from './table_vis_fn';
+import { TableVisApp } from './components/table_vis_app';
+
+export const getTableVisRenderer: (
+  core: CoreStart
+) => ExpressionRenderDefinition<TableVisRenderValue> = (core) => ({
+  name: 'table_vis',
+  displayName: 'table visualization',
+  reuseDomNode: true,
+  render: async (domNode, { visData, visConfig }, handlers) => {
+    handlers.onDestroy(() => {
+      unmountComponentAtNode(domNode);
+    });
+
+    const showNoResult = visData.table
+      ? visData.table.rows.length === 0
+      : visData.tableGroups?.length === 0;
+    render(
+      <VisualizationContainer className="tableVis" showNoResult={showNoResult}>
```

Comment:
I think it could be removed. I added it to make it consistent with other plugins like markdown, metrics and etc
the purpose I think is to avoid breaking if there is no table/tableGroups. but I don't see any test situation that this is possible. not sure if there is other design consideration.

Reply:
I'm okay keeping it, Just wanted to know what it did. Have a hunch that it might be to render the no data averrable screen when data is missing, but i could be wrong.

<img width="220" alt="Screen Shot 2022-11-02 at 2 15 41 PM" src="https://user-images.githubusercontent.com/20453492/199603965-09f075a4-7de4-48e5-9cb4-bf505c6abfaa.png">